dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -127,12 +127,14 @@ public GitHubSourceCodeRepo(String githubTokenUsername, String githubTokenConten
             builder.cache(DockstoreWebserviceApplication.getCache(null));
         }
         OkHttpClient build = builder.build();
-        ObsoleteUrlFactory obsoleteUrlFactory = new ObsoleteUrlFactory(build);
-
-        HttpConnector okHttp3Connector = new ImpatientHttpConnector(obsoleteUrlFactory::open);
+        // Must set the cache max age otherwise kohsuke assumes 0 which significantly slows down our GitHub requests. The GitHub API uses a cache max-age of 60 seconds, so use that
+        OkHttpGitHubConnector okHttp3Connector = new OkHttpGitHubConnector(build, GITHUB_DEFAULT_MAX_CACHE_AGE_SECONDS);
```

Comment:
Will this cause some changes on github to take up to `GITHUB_DEFAULT_MAX_CACHE_AGE_SECONDS` to propagate to the webservice?  For example, if the user refreshes, makes a quick tweak, and refreshes again, will the latter refresh not be visible to the webservice until it's more than the "max age" old?

Reply:
Yeah, that's correct. I tested on QA and this seems to be the case already because the change doesn't show up immediately. I registered a workflow, refreshed, modified the workflow, then kept refreshing until the change showed up. It took about 50 seconds for the change to show up.

Reply:
Recommend we consider a lower value for `GITHUB_DEFAULT_MAX_CACHE_AGE_SECONDS`.   I vaguely recall certain types of github changes propagating faster (during the topics testing, a change to the repo info took about 20 seconds to make it to the webservice, I think).  As long as the max age is multiple seconds, we probably get most of the caching advantage...

Reply:
I did some topics testing and the changes took longer than 20 seconds for me on qa. It ranged from 45 - 60 seconds. Maybe your topic wasn't cached/it was older than max-cache age so it fetched it from GitHub? 

Regardless, I changed the max cache age to 30 seconds